binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -3,12 +3,72 @@
 
 import filecmp
 import os
+import re
 import sys
 from argparse import ArgumentParser
+from typing import Iterable, Set
 
-from ..reporter import exception_reporting
+from ..reporter import announce_and_check_output, exception_reporting
 from ._parser import add_version_argument_to
 
+_keywords_pattern = re.compile('KEYWORDS="(?P<keywords>[^"]*)"')
+
+
+def _get_relevant_keywords_set_for(ebuild_content: str,
+                                   accept_keywords: Iterable[str]) -> Set[str]:
+    match = _keywords_pattern.search(ebuild_content)
+    if match is None:
+        # if the KEYWORDS variable is not found in the ebuild,
+        # we will assume the ebuild needs to be listed
+        ebuild_keywords = accept_keywords
+    else:
+        ebuild_keywords = match.group('keywords')
+        ebuild_keywords = [kw for kw in ebuild_keywords.split(" ") if kw]
+
+    # replace special keywords by the relevant keywords from the ebuild
+    if '**' in accept_keywords:
+        accept_keywords = ebuild_keywords
+    elif '*' in accept_keywords:
+        accept_keywords = [kw for kw in ebuild_keywords if not kw.startswith('~')]
+    elif '~*' in accept_keywords:
+        accept_keywords = [kw for kw in ebuild_keywords if kw.startswith('~')]
+
+    return set(accept_keywords) & set(ebuild_keywords)
+
+
+def _keywords_included_in_ebuild(accept_keywords: Iterable[str],
+                                 new_portdir_ebuild_file: str,
+                                 old_portdir_ebuild_file: str = None) -> bool:
+    """
+    Check if the new_portdir_ebuild_file contains the accept_keywords. Returns True if that is the
+    case, otherwise False. If old_portdir_ebuild_file is passed, an extra comparison is made
+    between the relevant keywords in the old and the new ebuild. If these keywords are identical,
+    return False.
+    """
+
+    with open(new_portdir_ebuild_file) as ifile:
+        new_ebuild_content = ifile.read()
+    new_ebuild_relevant_keywords = _get_relevant_keywords_set_for(new_ebuild_content,
+                                                                  accept_keywords)
+
+    # return False if the new file doesn't include the specified keywords
+    if not new_ebuild_relevant_keywords:
+        return False
+
+    if old_portdir_ebuild_file is not None and os.path.exists(old_portdir_ebuild_file):
+        with open(old_portdir_ebuild_file) as ifile:
+            old_ebuild_content = ifile.read()
+        old_ebuild_relevant_keywords = _get_relevant_keywords_set_for(
+            old_ebuild_content, accept_keywords)
+
+        # return False if both old and new file include the same keywords
+        # (i.e., when only other keywords have changed)
+        if new_ebuild_relevant_keywords == old_ebuild_relevant_keywords:
+            return False
```

Comment:
So we'd get here if
- the two files are not identical (or else the function would not have been called) and
- the set of extracted keywords are identical

That means we are dealing with changes outside of the keywords line and the person making the changes decided they are not worth a revbump, okay.
We effectively saying "keywords are not included in the ebuild" in this line — but they are. To me this is a sign that the semantics of this function are not to the point enough yet or a mismatch with the function name. Let's make semantics obvious.